### PR TITLE
Update type-converter.adoc

### PR DESCRIPTION
**Pull request by fg78nc**
Corrected documentation for automatic Type Converter class path discovery for Camel running on Spring Boot. Tested with Camel 3.7.0-SNAPSHOT.

### CODE REVIEW

File: docs/user-manual/modules/ROOT/pages/type-converter.adoc [hunk 1]
```diff
@@ -134,7 +134,7 @@ You can declare multiple `<bean>`s if you have more classes.
 Camel will automatic discover and load type converters from all the JARs in the classpath on startup.
 
 Camel will search the classpath for a file called
-`META-INF/services/org/apache/camel/TypeConverterLoader`, which lists
+`META-INF/services/org/apache/camel/TypeConverter`, which lists
```
Comment (davsclaus) [inline docs/user-manual/modules/ROOT/pages/type-converter.adoc:137]: This is wrong, it should be as it was before

**Review by davsclaus — CHANGES_REQUESTED** [review]

File: docs/user-manual/modules/ROOT/pages/type-converter.adoc (hunk 1 above)
Comment (fg78nc) [reply docs/user-manual/modules/ROOT/pages/type-converter.adoc:137]: With all due respect, please see verifiable example with unit test.
https://github.com/fg78nc/camel-type-converter

Comment (davsclaus) [reply docs/user-manual/modules/ROOT/pages/type-converter.adoc:137]: You are using the old way (camel 2.x style). In Camel 3.x there is a fast TypeConverterLoader that is recommended to use.

Comment (fg78nc) [reply docs/user-manual/modules/ROOT/pages/type-converter.adoc:137]: Where can I find relevant documentation for Camel 3.x?

Comment (davsclaus) [reply docs/user-manual/modules/ROOT/pages/type-converter.adoc:137]: You are looking at it - that doc file has specific sections about 2.x

Comment (fg78nc) [reply docs/user-manual/modules/ROOT/pages/type-converter.adoc:137]: `@Converter(loader = true)` -  I don't see this "loader" element in `@Converter `https://www.javadoc.io/doc/org.apache.camel/camel-api/latest/org/apache/camel/Converter.html

Comment (davsclaus) [reply docs/user-manual/modules/ROOT/pages/type-converter.adoc:137]: Ah yeah the option is renamed 

@Converter(generateLoader = true)

Can you fix that in the docs

Comment (fg78nc) [reply docs/user-manual/modules/ROOT/pages/type-converter.adoc:137]: I fixed the docs. Now with those changes I have working example for Camel 3.x. 
I still think documentation is scarce. Would you like to add more details to documentation and probably add working example to Camel examples?

Comment (davsclaus) [reply docs/user-manual/modules/ROOT/pages/type-converter.adoc:137]: Yeah an example is welcome with a custom type converter. The old way with loader disabled is still possible, but its slower and uses reflection at invoking the converter at runtime. The loader generates source code so its very fast.

Comment (davsclaus) [reply docs/user-manual/modules/ROOT/pages/type-converter.adoc:137]: You can tidy up this PR and we can merge it. Then you can do new PRs for more docs, and another with the example

Comment (fg78nc) [reply docs/user-manual/modules/ROOT/pages/type-converter.adoc]: Done, please review.

**Review by davsclaus — APPROVED** [review]

**Pull request merged**